jobs/integration/validation.py

```diff
@@ -2085,104 +2085,128 @@ async def test_multus(model, tools, addons_model):
-                    "image": "rocks.canonical.com/cdk/ubuntu:focal",
-                    "command": ["sleep", "3600"],
+                    "name": "alpine",
+                    "image": "rocks.canonical.com/alpine:latest",
+                    "command": [
+                        "/bin/ash",
+                        "-c",
+                        "trap : TERM INT; sleep 3600 & wait",
+                    ],
                 }
             ]
         },
     }
-    remote_path = "/tmp/pod.yaml"
-    with NamedTemporaryFile("w") as f:
-        yaml.dump(pod_definition, f)
-        await scp_to(f.name, unit, remote_path, tools.controller_name, tools.connection)
-    await run_until_success(
-        unit,
-        "/snap/bin/kubectl --kubeconfig /root/.kube/config apply -f " + remote_path,
-    )
+    await apply_def(attachnet_def)
+    await apply_def(pod_definition)
 
     # Verify pod has the expected interfaces
-    await run_until_success(
-        unit,
-        "/snap/bin/kubectl --kubeconfig /root/.kube/config exec multus-test -- apt update",
-    )
-    await run_until_success(
-        unit,
-        "/snap/bin/kubectl --kubeconfig /root/.kube/config exec multus-test -- apt install -y iproute2",
-    )
     output = await run_until_success(
         unit,
-        "/snap/bin/kubectl --kubeconfig /root/.kube/config exec multus-test -- ip addr",
+        "/snap/bin/kubectl --kubeconfig /root/.kube/config exec multus-test -- ip a",
     )
     # behold, ugly output parsing :(
-    lines = output.splitlines()
-    active_interfaces = set()
-    while lines:
-        line = lines.pop(0)
-        interface = line.split()[1].rstrip(":").split("@")[0]
-        while lines and lines[0].startswith(" "):
+    try:
+        lines = output.splitlines()
+        active_networks = dict()
+        while lines:
             line = lines.pop(0)
-            if line.split()[0] == "inet":
-                # interface has an address, we'll call that good enough
-                active_interfaces.add(interface)
-    expected_interfaces = ["eth0", "net1", "net2"]
-    for interface in expected_interfaces:
-        if interface not in active_interfaces:
-            pytest.fail(
-                "Interface %s is missing from ip addr output:\n%s" % (interface, output)
-            )
-
-    await cleanup()
+            interface = line.split()[1].rstrip(":").split("@")[0]
+            while lines and lines[0].startswith(" "):
+                line = lines.pop(0).strip()
+                if line.startswith("inet "):
+                    # interface has an address, we'll call that good enough
+                    active_networks[interface] = line.split()[1]
+        expected_interfaces = ["eth0", "net1"]
+        for ifc in expected_interfaces:
+            assert (
+                ifc in active_networks
+            ), f"Interface {ifc} is missing from ip addr output:\n{output}"
+            assert active_networks[ifc].startswith(
+                "10.166.0."
```

Comment:
It's a `/16` subnet so only the first two octets are fixed

```suggestion
                "10.166."
```